server/src/main/java/org/opensearch/cluster/coordination/JoinTaskExecutor.java

```diff
@@ -511,11 +511,26 @@ private static void ensureRemoteStoreNodesCompatibility(DiscoveryNode joiningNod
         assert existingNodes.isEmpty() == false;
 
         CompatibilityMode remoteStoreCompatibilityMode = REMOTE_STORE_COMPATIBILITY_MODE_SETTING.get(metadata.settings());
+        List<String> reposToSkip = new ArrayList<>();
+        // Skip checking for remote routing table repo if feature is not enabled.
+        if (!RemoteStoreNodeAttribute.isRemoteRoutingTableEnabled(metadata.settings())) {
+            String joiningNodeRepoName = joiningNode.getAttributes()
+                .get(RemoteStoreNodeAttribute.REMOTE_STORE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEY);
+            String existingNodeRepoName = existingNodes.get(0)
+                .getAttributes()
+                .get(RemoteStoreNodeAttribute.REMOTE_STORE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEY);
+            if (joiningNodeRepoName != null) {
+                reposToSkip.add(joiningNodeRepoName);
+            }
+            if (existingNodeRepoName != null) {
+                reposToSkip.add(existingNodeRepoName);
```

Comment:
Move this logic to directly to `RemoteStoreNodeAttribute`

Reply:
Done